File: marketplace-ui/src/app/core/models/order.ts
export enum OrderStatus {
  PENDING = 'PENDING',
  CONFIRMED = 'CONFIRMED',
  SHIPPED = 'SHIPPED',
  DELIVERED = 'DELIVERED',
  CANCELLED = 'CANCELLED',
}

export enum PaymentMethod {
  CASH_ON_DELIVERY = 'CASH_ON_DELIVERY',
}

export interface OrderItem {
  productId: string;
  productName: string;
  sellerId: string | null;
  price: number;
  quantity: number;
  totalPrice: number;
}

export interface Order {
  id: string;
  userId: string;
  status: OrderStatus;
  totalAmount: number;
  fullName: string;
  address: string;
  city: string;
  postalCode: string;
  phoneNumber: string;
  createdAt: number;
  cartItems: OrderItem[];
  paymentMethod: PaymentMethod;
}

export interface ShippingAddressRequest {
  fullName: string;
  address: string;
  city: string;
  postalCode: string;
  phone: string;
}

export interface CreateOrderRequest {
  shippingAddress: ShippingAddressRequest;
  paymentMethod: PaymentMethod;
}

File: Backend/orders/src/main/java/service/orders/models/Order.java
package service.orders.models;

import java.util.List;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Document(collection = "orders")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Order {
    @Id
    private String id;

    @Field("user_id")
    private String userId;

    @Field("status")
    private OrderStatus status;

    @Field("total_Amount")
    private Double totalAmount;

    @Field("full_name")
    private String fullName;

    @Field("address")
    private String address;

    @Field("City")
    private String city;

    @Field("postal_code")
    private String postalCode;

    @Field("phone_number")
    private String phoneNumber;

    @Field("created_at")
    private Long createdAt;

    @Field("cart_items")
    private List<CartItems> cartItems;

    @Field("payment_method")
    private PaymentMethod paymentMethod;
    


}